docs/bench_repo_specification.md

```diff
@@ -30,27 +30,53 @@ Example:
 ./bench ../my_little_compiler/
 ```
 
-| Return code | Explanation |
-|-------------|-------------|
-| 0 | No error |
-| 1 | Internal script error |
-| 2 | Incorrect script usage |
+| Return code | Explanation            |
+|-------------|------------------------|
+| 0           | No error               |
+| 1           | Internal script error  |
+| 2           | Incorrect script usage |
 
 The script should output its results on `stdout`.
 The result is encoded in JSON.
-Output to `stderr` is shown only while benchmarking and in case of an error.
+Output to `stderr` is shown only while benchmarking (streamed to the task-detail page) and in case of an error.
 
 ### Output format
 
-If an error occurs that prevented the benchmark script from running successfully, this format should be used:
+Executing the `bench` script isn't always successful and VelCom groups the
+errors into two categories:
+- `bench` script panics, which indicate that the benchmark script itself
+  crashed (i.e. the author of the benchmark repo needs to fix a bug).  
+  Example: The `bench` script suffered a segmentation fault.
+- errors of the benchmarked program, which indicate that the `bench` script
+  worked correctly, but the revision it tested had a critical problem (i.e. the
+  program under test has a problem and needs to be fixed).  
+  Example: The program under test fails to compile and therefore the bench
+  `script` can not test it.
+
+Benchmark script panics are indicated by a non-zero exit code of the `bench`
+script. If this happens, VelCom tries to provide as much information as it can
+in the UI, including the full `stdout`, `stderr`, information about the
+execution environment and maybe a potential cause for the error. Panics always
+take precedence over errors of the benchmarked program: If the `bench` script
+exits with a non-zero exit code, it is *always* treated as a benchmark script
+panic.
+
```

Comment:
Maybe move `### Output format` here and instead use a different heading for the preceding section, e.g. `### Failure modes`?